solcjs

```diff
@@ -15,16 +15,31 @@ process.removeAllListeners('uncaughtException');
 var commander = require('commander');
 
 const program = new commander.Command();
+const commanderParseInt = function(value) {
+  const parsedValue = parseInt(value, 10);
```

Comment:
Looks like [`parseInt()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt) is **very** lax when parsing integers:
> If `parseInt` encounters a character that is not a numeral in the specified `radix`, it ignores it and all succeeding characters and returns the integer value parsed up to that point. `parseInt` truncates numbers to integer values. Leading and trailing spaces are allowed.

So something like this
```bash
./solcjs /tmp/test.sol --bin --optimize-runs 6Y7Z8___sdfas
```
is accepted and the value is `6`.

I think we would have been better off with the original version that just passed the value to solc directly. But I won't be changing this back now. I'd like to get this finally merged.